include/logger.hrl

```diff
@@ -18,33 +18,55 @@
 
 -compile({parse_transform, emqx_logger}).
 
+%% deprecated
 -define(DEBUG(Format), ?LOG(debug, Format, [])).
 -define(DEBUG(Format, Args), ?LOG(debug, Format, Args)).
 
+%% deprecated
 -define(INFO(Format), ?LOG(info, Format, [])).
 -define(INFO(Format, Args), ?LOG(info, Format, Args)).
 
+%% deprecated
 -define(NOTICE(Format), ?LOG(notice, Format, [])).
 -define(NOTICE(Format, Args), ?LOG(notice, Format, Args)).
 
+%% deprecated
 -define(WARN(Format), ?LOG(warning, Format, [])).
 -define(WARN(Format, Args), ?LOG(warning, Format, Args)).
 
+%% deprecated
 -define(ERROR(Format), ?LOG(error, Format, [])).
 -define(ERROR(Format, Args), ?LOG(error, Format, Args)).
 
+%% deprecated
 -define(CRITICAL(Format), ?LOG(critical, Format, [])).
 -define(CRITICAL(Format, Args), ?LOG(critical, Format, Args)).
 
+%% deprecated
 -define(ALERT(Format), ?LOG(alert, Format, [])).
 -define(ALERT(Format, Args), ?LOG(alert, Format, Args)).
 
+%% deprecated
 -define(LOG(Level, Format), ?LOG(Level, Format, [])).
 
+%% deprecated
 -define(LOG(Level, Format, Args),
         begin
           (logger:log(Level,#{},#{report_cb => fun(_) -> {'$logger_header'()++(Format), (Args)} end,
                                   mfa => {?MODULE, ?FUNCTION_NAME, ?FUNCTION_ARITY},
                                   line => ?LINE}))
         end).
 
+%% structured logging
+-define(SLOG(Level, Text, DataExpr), % DataExpr must evaluate to a map(), better to be JSON friendly
+        case logger:allow(Level, ?MODULE) of
+            true ->
+                logger:log(Level,
+                           (DataExpr)#{ labels => [iolist_to_binary(I) || I <- string:tokens('$logger_header'(), "[ ]")]
```

Comment:
I think it should go to the metadata instead

Reply:
Not sure how useful it is given they are merged to create json object anyway

Reply:
This merging should perhaps happen in the logger formatter

Reply:
will address this in a separate PR.